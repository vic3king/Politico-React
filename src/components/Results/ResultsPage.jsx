import React, { Component } from 'react';
import Notifications from 'react-notify-toast';
import Offices from '../../services/offices';
import ResultsModal from '../Modals/ResultModal';
import caputalizeString from '../../helpers/capitalizer';
import Button from '../shared/Buttons/Button';
import Loader from '../shared/Loader/Loader';

class ResultsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      offices: [],
      loading: true,
      showResultsModal: false,
      officeId: null,
    };
  }

  componentDidMount = async () => {
    const allOffices = await Offices.getAllOffices();
    this.setState({ offices: allOffices.data, loading: false });
  };

  showOfficeModal = e => {
    this.setState({ showResultsModal: true, officeId: e.target.id });
  };

  hideResultsModal = () => {
    this.setState({ showResultsModal: false });
  };

  render() {
    const { offices, loading, showResultsModal, officeId } = this.state;

    const listOfOffices = offices.map(office => (
      <div key={office.id} className="card">
        <h6>
          {caputalizeString(office.name)}({caputalizeString(office.type)})
        </h6>
        <p className="num">0</p>
        <Button
          className="admin-office-card"
          id={office.id}
          value="View Result"
          onClick={this.showOfficeModal}
        />
      </div>
    ));

    return (
      <React.Fragment>
        <Notifications />
        {loading && <Loader />}
        <h4 className="page-title-results">Election Results</h4>
        {showResultsModal && (
          <ResultsModal hide={this.hideResultsModal} officeId={officeId} />
        )}
        <div id="grid">{listOfOffices}</div>
      </React.Fragment>
    );
  }
}

export default ResultsPage;
