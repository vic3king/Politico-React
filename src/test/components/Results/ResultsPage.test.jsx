import React from 'react';
import { shallow } from 'enzyme';
import ResultsPage from '../../../components/Results/ResultsPage';

import Offices from '../../../services/offices';

jest.mock('../../../services/offices');

let wrapper;

describe('ResultsPage', () => {
  beforeAll(() => {
    Offices.getAllOffices = jest.fn().mockImplementation(() =>
      Promise.resolve({
        status: 200,
        data: [
          {
            name: '',
          },
          {
            name: '',
          },
        ],
      })
    );
  });

  beforeEach(() => {
    wrapper = shallow(<ResultsPage />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('ComponentDidMount', () => {
    it('should call componentDidMount method', async () => {
      const instance = wrapper.instance();
      await instance.componentDidMount();

      expect(instance.state.offices).toEqual([
        {
          name: '',
        },
        {
          name: '',
        },
      ]);
    });
  });

  // describe('showOfficeModal method', () => {
  //   it('should set showResultsModal to true and get and ID', () => {
  //     const instance = wrapper.instance();
  //     const e = {
  //       target: { id: 1 },
  //     };

  //     instance.setState({
  //       showResultsModal: true,
  //     });

  //     instance.showOfficeModal(e);

  //     expect(instance.state.showResultsModal).toBeTruthy();
  //     expect(instance.state.officeId).toBe(2);
  //   });
  // });

  describe('hideResultsModal method', () => {
    it('should set showResultsModal to false ', () => {
      const instance = wrapper.instance();
      instance.setState({
        showResultsModal: false,
      });

      instance.hideResultsModal();

      expect(instance.state.showResultsModal).toBeFalsy();
    });
  });
});
