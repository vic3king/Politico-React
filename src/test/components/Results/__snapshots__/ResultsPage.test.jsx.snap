// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <_class
          options={Object {}}
        />,
        <Loader />,
        <NavBar
          LiTagOne={
            <LiTag
              to="/logout"
              value="Logout"
            />
          }
          LiTagTwo=""
        />,
        <h4
          className="page-title-results"
        >
          Election Results
        </h4>,
        false,
        <div
          id="grid"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "options": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "LiTagOne": <LiTag
            to="/logout"
            value="Logout"
          />,
          "LiTagTwo": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Election Results",
          "className": "page-title-results",
        },
        "ref": null,
        "rendered": "Election Results",
        "type": "h4",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "id": "grid",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <_class
            options={Object {}}
          />,
          <Loader />,
          <NavBar
            LiTagOne={
              <LiTag
                to="/logout"
                value="Logout"
              />
            }
            LiTagTwo=""
          />,
          <h4
            className="page-title-results"
          >
            Election Results
          </h4>,
          false,
          <div
            id="grid"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "options": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "LiTagOne": <LiTag
              to="/logout"
              value="Logout"
            />,
            "LiTagTwo": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Election Results",
            "className": "page-title-results",
          },
          "ref": null,
          "rendered": "Election Results",
          "type": "h4",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "id": "grid",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
